### PR TITLE
Feat/support docker binaries

### PR DESCRIPTION
**Pull request by orkamara**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Add the new binaries issues into the output of the CLI

#### Any background context you want to provide?
The change support both regular output + `--json` flag
We had to change some logic around `introducedThrough` and `fromPaths` to make it work.

#### What are the relevant tickets?
https://snyksec.atlassian.net/secure/RapidBoard.jspa?projectKey=DC&rapidView=62&selectedIssue=DC-7

### CODE REVIEW

File: src/cli/commands/test.js [hunk 1]
```diff
@@ -557,7 +565,7 @@ function groupVulnerabilities(vulns) {
       map[curr.id].severity = curr.severity;
       map[curr.id].isNew = isNewVuln(curr);
     }
-    if (curr.upgradePath) {
+    if (curr.upgradePath && curr.upgradePath[0]) {
```
Comment (adrukh) [inline src/cli/commands/test.js]: What is the purpose of this change?

Comment (orkamara) [reply src/cli/commands/test.js]: It's basically a big fix - we didn't check if the list is empty - only if its declared.
The result is that we might give the user remediation advice even if `upgradePath.length == 0`.
The reason why we didn't discover the bug before is because the check `curr.upgradePath[1] === curr.from[1];` - but in our case those 2 lists are empty

Comment (adrukh) [reply src/cli/commands/test.js]: Hmmmm seems really important! Is this on a separate commit? If it doesn’t break the new flow, maybe it deserves its own test, PR and release?

Comment (orkamara) [reply src/cli/commands/test.js]: changed it - need to invest more and figure out how to test it
